infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,475 @@
+package org.corfudb.infrastructure.configuration;
+
+
+import io.netty.channel.EventLoopGroup;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.corfudb.comm.ChannelImplementation;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * This class is holds various server configuration parameters.
+ *
+ * <p>Created by maithem on 12/4/19.
+ */
+
+public class ServerConfiguration extends PropertiesConfiguration {
+    // Server general parameters
+    static final String SERVER_DIR = "serverDirectory";
+    static final String SINGLE_MODE = "singleMode";
+    static final String CLUSTER_ID = "clusterId";
+    static final String NUM_IO_THREADS = "numIOThreads";
+    static final String HOST_ADDRESS = "hostAddress";
+    static final String SERVER_PORT = "serverPort";
+    static final String NETWORK_INTERFACE = "networkInterface";
+    static final String HANDSHAKE_TIMEOUT = "handshakeTimeout";
+    static final String METADATA_RETENTION = "metadataRetention";
+    static final String LOG_LEVEL = "logLevel";
+    static final String NUM_BASE_SERVER_THREADS = "numBaseServerThreads";
+    static final String METRICS_PROVIDER_ADDRESS = "metricsProviderAddress";
+    static final String CHANNEL_IMPLEMENTATION = "channelImplementation";
+    static final String ENABLE_TLS = "enableTls";
+    static final String ENABLE_TLS_MUTUAL_AUTH = "enableTlsMutualAuth";
+    static final String KEYSTORE = "keystore";
+    static final String KEYSTORE_PASSWORD_FILE = "keystorePasswordFile";
+    static final String TRUSTSTORE = "truststore";
+    static final String TRUSTSTORE_PASSWORD_FILE = "truststorePasswordFile";
+    static final String ENABLE_SASL_PLAIN_TEXT_AUTH = "enableSaslPlainTextAuth";
+    static final String SASL_PLAIN_TEXT_USERNAME_FILE = "saslPlainTextUsernameFile";
+    static final String SASL_PLAIN_TEXT_PASSWORD_FILE = "saslPlainTextPasswordFile";
+    static final String TLS_CIPHERS = "tlsCiphers";
+    static final String TLS_PROTOCOLS = "tlsProtocols";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651442)